ddp-workspace/projects/ddp-singular/src/app/components/activity/activity.component.ts

```diff
@@ -15,6 +15,14 @@ export class ActivityComponent extends ActivityRedesignedComponent {
   Route = Route;
   isCaptchaResolved = false;
 
+  @HostListener('document:keydown.enter', ['$event']) onEnter(event:KeyboardEvent) {
+      super.nextField(event, {
```

Comment:
**_#optional_** 
Should the event fire only on input fields ?

E.g. a corner case (rare and maybe weird, but anyway): 
 - there 10 input fields in a form
 - a user types something in 5th input field
 - the user moves focus out of the edited field (click on the document outside, press Tab and focus on a next button, 
    etc.)
- the user presses Enter by accident
- what field has focused ?

Reply:
Not only, but there might also be another type of Elements, like 'textarea' or 'select'.
Each time the user moves out focus, after pressing enter - the first field will be focused.

Reply:
Btw, what about a default behavior by pressing Enter: 
- **textarea** - go to the next line
- **select**  open a dropdown options list (even though it can be done by Space or mouse click) ?

It will be unavailable, won't it ?

Reply:
Yes, right, but as they asked, I had to change the default behavior of pressing Enter. I think that I probably could find some workaround, though. What do you say?

Reply:
I do agree that it changes the default HTML elements behavior, but is it not what was asked in this following task: 
- If you hit "return" on your keyboard, it takes you back to the dashboard and you have to start over. Hitting return should take you to the next line of the form

Yes, I did so only because I expected the same functionality would be required by other studies as well, and they could just easily call the same function. I also thought about implementing this functionality differently, though.

Reply:
> default HTML elements behavior

I mean e.g. go to the next line in a textarea. 

>  hit "return" on your keyboard, it takes you back to the dashboard and you have to start over

Most likely it is an unknown (?) bug. At least a behavior that nobody wanted to fix before.
I don't think it is a special feature.

Reply:
@MocanaAtBroad, what are your thoughts about it ?

Reply:
> I mean e.g. go to the next line in a textarea.

Yes, I got this and I thought the same about changing default behavior. I have even special option:
```
export interface NavigateFieldsOptionsModel {
    elements: string[];
    preventDefault?: boolean; <---- this
    carousel?: boolean;
}
```

> Most likely it is an unknown (?) bug. At least a behavior that nobody wanted to fix before. I don't think it is a special feature.

What do you think about only fixing this bug and keeping the default behavior of the enter key?

Reply:
Okay. Btw, We could also just avoid preventing default behaviour only with Textarea and Select HTML elements. I think input fields are fine. But as you said, it must be discussed.